firebase-firestore/src/main/java/com/google/firebase/firestore/FirebaseFirestore.java

```diff
@@ -193,6 +212,11 @@ public void setFirestoreSettings(@NonNull FirebaseFirestoreSettings settings) {
                 + "You can only call setFirestoreSettings() before calling any other methods on a "
                 + "FirebaseFirestore object.");
       }
+
+      if (this.emulatorSettings != null) {
```

Comment:
This merge needs to happen before line 209, otherwise the "equals" check will fail even if the user calls this method with the same settings.

Also nit: Drop "this"